snuba/datasets/entities/sessions.py

```diff
@@ -211,7 +212,18 @@ def __init__(self) -> None:
     def select_storage(
         self, query: Query, request_settings: RequestSettings
     ) -> StorageAndMappers:
-        granularity = extract_granularity_from_query(query, "started") or 3600
+
+        granularity = None
+        # This checks if we are creating this query as a consequence of a Crash Rate Alert and if so
+        # we set the granularity sent by Sentry.
+        # For time windows <= 1h, we expect to use granularity of 60s, and 3600s for time windows
+        # greater than that
+        for col in query.get_selected_columns():
+            if col.name and bool(CRASH_RATE_ALERT_PATTERN.match(col.name)):
+                granularity = query.get_granularity()
+
+        if granularity is None:
+            granularity = extract_granularity_from_query(query, "started") or 3600
```

Comment:
Why can't we just use the granularity specified in the query to begin with rather than infer it from some random alias used in the query? Under what scenarios would we expect those values to be different anyway and why?

Reply:
granularity is not a reliable indicator at this point of the query in the ast + snql world with entities. 
If a query processor on the entity makes changes to the time condition, granularity is not kept up to date. Te only source of truth is the actual timestamp condition.
The entire concept of query granaularity should be rethought.

Reply:
So I set the granularity now in the query processor, and I do not call the `extract_granularity_from_query` if i did set it and it is for a crash rate alert subscription.. is that a good option

Reply:
The `granularity` field is not a source of truth, since the `TimeSeriesProcessor` will change the conditions of the query but not the granularity field. Is there a reason to change this at all? Why not extract the granularity here?